tests/utils/utils.go

```diff
@@ -0,0 +1,335 @@
+func RelayMessage(
+	ctx context.Context,
+	sourceBlockHash common.Hash,
+	sourceBlockNumber *big.Int,
+	source SubnetTestInfo,
+	destination SubnetTestInfo,
+) {
+	log.Info("Fetching relevant warp logs from the newly produced block")
+	logs, err := source.ChainWSClient.FilterLogs(ctx, interfaces.FilterQuery{
+		BlockHash: &sourceBlockHash,
+		Addresses: []common.Address{warp.Module.Address},
+	})
+	Expect(err).Should(BeNil())
+	Expect(len(logs)).Should(Equal(1))
+
+	// Check for relevant warp log from subscription and ensure that it matches
+	// the log extracted from the last block.
+	txLog := logs[0]
+	log.Info("Parsing logData as unsigned warp message")
+	unsignedMsg, err := avalancheWarp.ParseUnsignedMessage(txLog.Data)
+	Expect(err).Should(BeNil())
+
+	// Set local variables for the duration of the test
+	unsignedWarpMessageID := unsignedMsg.ID()
+	unsignedWarpMsg := unsignedMsg
+	log.Info("Parsed unsignedWarpMsg", "unsignedWarpMessageID", unsignedWarpMessageID, "unsignedWarpMessage", unsignedWarpMsg)
+
+	// Loop over each client on chain A to ensure they all have time to accept the block.
+	// Note: if we did not confirm this here, the next stage could be racy since it assumes every node
+	// has accepted the block.
+	WaitForAllValidatorsToAcceptBlock(ctx, source.ChainNodeURIs, source.BlockchainID, sourceBlockNumber.Uint64())
+
+	// Get the aggregate signature for the Warp message
+	log.Info("Fetching aggregate signature from the source chain validators")
+	warpClient, err := warpBackend.NewClient(source.ChainNodeURIs[0], source.BlockchainID.String())
+	Expect(err).Should(BeNil())
+	signedWarpMessageBytes, err := warpClient.GetAggregateSignature(ctx, unsignedWarpMessageID, params.WarpQuorumDenominator)
+	Expect(err).Should(BeNil())
+
+	// Construct the transaction to send the Warp message to the destination chain
+	signedTx := CreateReceiveCrossChainMessageTransaction(
+		ctx,
+		signedWarpMessageBytes,
+		big.NewInt(1),
+		teleporterContractAddress,
+		fundedAddress,
+		fundedKey,
+		destination.ChainWSClient,
+		destination.ChainIDInt,
+	)
+
+	log.Info("Sending transaction to destination chain")
+	receipt := SendTransactionAndWaitForAcceptance(ctx, destination.ChainWSClient, signedTx)
+
+	receiveCrossChainMessageLog := receipt.Logs[0]
+	var event ReceiveCrossChainMessageEvent
+	err = teleporter.EVMTeleporterContractABI.UnpackIntoInterface(&event, "ReceiveCrossChainMessage", receiveCrossChainMessageLog.Data)
+	Expect(err).Should(BeNil())
+}
+
+//
+// Unexported functions
+//
+
+func signTransaction(tx *types.Transaction, key *ecdsa.PrivateKey, chainID *big.Int) *types.Transaction {
+	txSigner := types.LatestSignerForChainID(chainID)
+	signedTx, err := types.SignTx(tx, txSigner, key)
+	Expect(err).Should(BeNil())
+
+	return signedTx
+}
+
+func calculateTxParams(ctx context.Context, wsClient ethclient.Client, fundedAddress common.Address) (*big.Int, *big.Int, uint64) {
```

Comment:
Nit: Can you add a comment above this function documenting what the return values are.

Reply:
Added